compiler/rustc_expand/src/expand.rs

```diff
@@ -1225,7 +1269,9 @@ impl<'a, 'b> MutVisitor for InvocationCollector<'a, 'b> {
     fn visit_pat(&mut self, pat: &mut P<ast::Pat>) {
         match pat.kind {
             PatKind::MacCall(_) => {}
-            _ => return noop_visit_pat(pat, self),
+            _ => {
+                return assign_id!(self, &mut pat.id, || noop_visit_pat(pat, self));
+            }
```

Comment:
This shouldn't be necessary because patterns do not support attributes (https://github.com/rust-lang/rust/pull/87146#issuecomment-881890210).